server/cluster/cluster.go

```diff
@@ -914,9 +922,140 @@ func (c *RaftCluster) PutStore(store *metapb.Store) error {
 	}
 	c.OnStoreVersionChange()
 	c.AddStoreLimit(store)
+	go recordUpProgress(c, store, progressMaxRetryLimit)
 	return nil
 }
 
+// recordUpProgress record the progress of new store.
+//It will stop recording if region count not change continue 100 or progress is more than 90
+func recordUpProgress(rc *RaftCluster, src *metapb.Store, maxRetryLimit int) storeProgressStatus {
+	progress := 0
+	regionCount := 0
+	retryTime := 0
+	storeId := strconv.FormatUint(src.GetId(), 10)
+	targetStore := pickSimilarStore(rc.GetStores(), src)
+
+	for {
+		if progress > 90 {
+			storeProgressGauge.WithLabelValues(src.Address, storeId, "up").Set(100.00)
```

Comment:
Will it cause a sudden change from 90 to 100?

Reply:
yes  , I think if  the region count more than 90 percent of the target store ,it will finish